### DISCUSSION TITLE
In the same link, if request 1 blocks, will all subsequent requests block? Is request2 affected by request1?

### DISCUSSION THREAD

**Discussion by jishengming1** [apache/kvrocks#1243] [Q&A]
Hi,@git-hulk I have a question for help.
In the same link, if request 1 blocks, will all subsequent requests block? Is request2 affected by request1?

**Comment by git-hulk** [comment]
Yes, that's right.

**Reply by jishengming1** [reply]
Got it, thanks a lot!

**Comment by xiaobiaozhao** [comment]
Yes, you can think of this as an http1.1 pipeline

**Reply by jishengming1** [reply]
Got it, thanks a lot!

**Comment by jishengming1** [comment]
@xiaobiaozhao hi，Please ask again, how to get the total number of keys. 
I used the info keyspace command and the result:  db0: keys=0,expires=0,avg_ttl=0,expired=0

**Reply by git-hulk** [reply]
You should use `dbsize scan` to async scan the database first, then use info keyspace to have a look again after finishing.

**Reply by jishengming1** [reply]
> You should use `dbsize scan` to async scan the database first, then use info keyspace to have a look again after finishing.
Okay, I'll try, thanks!

**Reply by git-hulk** [reply]
The main reason is Kvrocks stores its data in disk instead of memory, so can't just simply keep the key counter in memory. And it's impossible to store the row count like MySQL since we supports key expire feature, so we can't know how many keys are expired before scanning the database.

**Reply by jishengming1** [reply]
> The main reason is Kvrocks stores its data in disk instead of memory, so can't just simply keep the key counter in memory. And it's impossible to store the row count like MySQL since we supports key expire feature, so we can't know how many keys are expired before scanning the database.

Got it, thanks a lot!

**Comment by jishengming1** [comment]
@xiaobiaozhao hi, Does it support prefix lookups?
 Currently used scan cursor  MATCH pattern COUNT count .  Get the key first and then get the value . But this requires two queries. is there a way to query only once?
